Marketplace/voiceHUDApp/App Resources/appData/voiceHUD_app.js

```diff
@@ -0,0 +1,815 @@
+            // of updating (the latter for efficiency too).
+            var audioLevel = 0.0;
+            var avgAudioLevel = 0.0;
+
+            if (user) {
+                // we will do exponential moving average by taking some the last loudness and averaging
+                user.accumulatedLevel = AVERAGING_RATIO * (user.accumulatedLevel || 0) + (1 - AVERAGING_RATIO) * (user.audioLoudness);
+
+                // add 1 to insure we don't go log() and hit -infinity.  Math.log is
+                // natural log, so to get log base 2, just divide by ln(2).
+                audioLevel = scaleAudio(Math.log(user.accumulatedLevel + 1) / LOG2);
+
+                // decay avgAudioLevel
+                avgAudioLevel = Math.max((1 - AUDIO_PEAK_DECAY) * (user.avgAudioLevel || 0), audioLevel).toFixed(3);
+
+            }
+
+            userStore[uuid].audioLevel = audioLevel;
+            userStore[uuid].avgAudioLevel = avgAudioLevel;
+        }
+    };
+
+    var lists = {
+        // returns an array of avatarPaldata
+        allAvatars: function () {
+            return AvatarManager.getPalData().data;
+        },
+        
+        // searches the top 10 list to get the index
+        getIndexOfSettingsUser: function (uuid) {
+            if (settings.users.length) {
+                var index = settings.users.map(function (item) {
+                    return item.uuid;
+                }).indexOf(uuid);
+                return index;
+            }
+            return -1;
+        },
+
+        sortData: function () {
+            // sort by audioLevel
+            function sortNumber(a, b) {
+                return b.avgAudioLevel - a.avgAudioLevel;
+            }
+            var avatarList = Object.keys(userStore);// : lists.getAvatarsInRadius(SEARCH_RADIUS);
+            settings.users = avatarList.map(function (uuid) { return userStore[uuid]; });
+        }
+    };
+
+    // overlay variables
+    var AVATAR_HEIGHT_OFFSET = 1.5,
+        HUD_MIN_Z_DIMENSIONS = 0.01,
+        HUD_MAX_Z_DIMENSIONS = [
+            0.1445,
+            0.2451,
+            0.3669,
+            0.1980,
+            0.0911
+        ],
+        HUD_ALPHA_START = 0.5,
+        BAR_ALPHA_START = 1,
+        HUD_DEFAULT_DIMENSIONS = { x: 0.05, y: 0.001, z: 0.01 },
+        hudIntervalTransparency,
+        hudIntervalTimeout;
+
+    var hud = {
+        // Create a HUD element over a user's head 
+        createHUD: function (uuid) {
+            console.log("adding huds");
+
+            var user = userStore[uuid];
+            user.hudID = [];
+            user.audioBarIDs = [];
+            user.userHeight = AvatarManager.getAvatar(uuid).getNeckPosition().y + AvatarManager.getAvatar(uuid).scale * AVATAR_HEIGHT_OFFSET;
+            var hudPosition = {x: AvatarManager.getAvatar(uuid).position.x, y: user.userHeight, z:AvatarManager.getAvatar(uuid).position.z}; // user.currentPosition
```

Comment:
just a reminder, another way you can write x,y,z now is [0,2,0] for a short hand if you ever need